vars/toolValidate.groovy

```diff
@@ -1,31 +1,45 @@
 import com.sap.piper.FileUtils
 import com.sap.piper.Version
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationLoader
+
 import hudson.AbortException
 
 
 def call(Map parameters = [:]) {
 
-    handlePipelineStepErrors (stepName: 'toolValidate', stepParameters: parameters) {
+    def stepName = 'toolValidate'
 
-        def tool = parameters.tool
-        def home = parameters.home
+    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters) {
+
+        final script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
-        if (!tool) throw new IllegalArgumentException("The parameter 'tool' can not be null or empty.")
-        if (!home) throw new IllegalArgumentException("The parameter 'home' can not be null or empty.")
+        final Map configuration = ConfigurationLoader.generalConfiguration(script)
+
+        def utils = new Utils()
+        def tool = utils.getMandatoryParameter(parameters, 'tool')
+        def home = parameters.home
 
-        FileUtils.validateDirectoryIsNotEmpty(home)
+        if (home) FileUtils.validateDirectoryIsNotEmpty(home)
 
         switch(tool) {
-            case 'java': validateJava(home)
-            return
-            case 'mta': validateMta(home)
-            return
-            case 'neo': validateNeo(home)
-            return
-            case 'cm': validateCm(home)
-            return
+            case 'java':
+                validateJava(home)
+                return
+            case 'mta':
+                if (home) validateMta("$home/mta.jar")
+                else validateMta(utils.getMtaJar(this, stepName, configuration, env))
```

Comment:
here `toolValidate` reuse the methods that are refactored.

Reply:
The if/else here looks like a smell. But I did not dive into the details. Maybe some explanation why it is not possible to have one call would be appropriate. Applies also below.

Reply:
see the description of the pull request, there it is explained